test/schema_generators/complete_test.cljc

```diff
@@ -19,6 +19,13 @@
       (is (= "test" (complete/complete "test" s)))
       (is (integer? (:foo (complete/complete {} s)))))))
 
+(deftest optional-key-complete-test
+  (let [s {:a s/Int (s/optional-key :b) s/Str}
+        ex-without-optional-keys (complete/complete {} s {} {} {} false)
+        ex-including-optional-keys (complete/complete {} s {} {} {} true)]
+    (is (nil? (:b ex-without-optional-keys)))
+    (is (string? (:b ex-including-optional-keys)))))
```

Comment:
This test is pretty basic; can you think of any cases that might not be handled here?

Reply:
It's been a long time.  might be worth testing other record-like things?

Reply:
Good call, I'll get a few of those in there.
I also wrote a test for nested schemas and it uncovered a related issue with my implementation. I'll try to tackle it in the next few days.